### PR TITLE
Fixed the Vm's `#reset` and impoved some methods and constructors

### PR DESCRIPTION
**Pull request by Jens0512**
The one method i forgot to make specs for had of course to be buggy.
`VM#reset` was raising an error. Anyways fixed now, together with specs and some minor changes to the vm

### CODE REVIEW

File: spec/vm_spec.cr [hunk 1]
```diff
@@ -134,5 +136,45 @@ describe "VM -" do
       vm.print_ast output
       output.to_s.should eq "Expressions\nCall\nConst\nStringLiteral|Hello world!\n"
     end
+
+    describe "#run(*programs)" do
+      it "takes files, filepaths or IOs; and (attempts to) run them" do
```
Comment (faultyserver) [inline spec/vm_spec.cr:141]: just for completeness, could you add a spec around what happens when one of the files/filepaths/IOs _can't_ be run? What should the expected behavior be?

File: src/myst/vm.cr [hunk 2]
```diff
@@ -6,12 +6,12 @@ module Myst
     # Instead `for_file` and `for_content` should be used
     # Many of these parameters would be rarely used
     # All having default values, i don't really think the amount of parameters is a problem
-    def initialize(source : IO = IO::Memory.new, source_name : String = "eval_input", with_stdlib? : Bool = true,
-                   use_stdios? : Bool = false, product : Bool = true)            
+    def initialize(source : IO = IO::Memory.new, *, source_name : String = "eval_input", with_stdlib? : Bool = true,
+                   use_stdios? : Bool = false, product? : Bool = true)
```
Comment (faultyserver) [inline src/myst/vm.cr:10]: what's the `*` parameter used for? Just to indicate that the remaining parameters must be passed by name?

**Comment by faultyserver** [comment]
Solid work! Just the two questions, then I'll gladly merge :)

File: src/myst/vm.cr (hunk 2 above)
Comment (Jens0512) [reply src/myst/vm.cr:10]: Yes

**Comment by faultyserver** [comment]
Changes look good to me! Thanks for this :)

**Pull request merged**